### DISCUSSION TITLE
Add a checked temp ring buffer for debugging.

### DISCUSSION THREAD

**Discussion by lerno** [odin-lang/Odin#1922] [Proposals]
The temp ring buffer may overrun without any way to detect it. This proposal details a debug ring buffer which which can catch use after "expected free":

1. If the ring buffer is usually n kb big, allocate a ring buffer n * 2 kb big.
2. Start by making the whole buffer read/write protected. If using virtual memory, then this may be done by flags, otherwise this will need to be added to ptr derefs.
X = invalid O = valid `[XXXX | XXXX]` (imagine a normal buffer as `[OOOO]`)
3. As allocations increase, start making the used regions valid: `[OOOX|XXXX]`
4. When the ring buffer start has passed, start invalidating the beginning of the buffer: `[XOOO|OXXX]`
5. This creates a moving valid segment that is n kb wide.
6. To detect even later use after free, increase this debug ring buffer to * 3 or * 4

**Comment by sumofat** [comment]
Would be useful
